Expand support for `--rgb`

melissalinkert opened the pull request:
Attempts to address #43, as discussed briefly with @sbesson. This still requires that the input channel count be a multiple of 3, as channel merging happens in groups of 3. As noted in the comments, this is maybe something we could consider making configurable as a next step.

More than one timepoint and/or Z section should be supported, along with channel counts greater than 3 (but still a multiple of 3). I tested with 12 channel Zeiss ZVI data (`curated/zeiss-zvi/cecilia/*.zvi`) and various RGB Z stacks/timelapse movies (e.g. some of `curated/apng/samples/`, `curated/zeiss-czi/qa-17173`).

sbesson reviewed (changes requested):
Tested with the sample files suggested in the description which were all converted using `bioformats2raw 0.5.0` aa NGFF then converted to OME-TIFF using either `raw2ometiff 0.3.1` or this PR with the `--rgb` flag

- with `raw2ometiff 0.3.1`, all converted files are stored as non RGB (`Channel.SamplesPerPixel=1` in the OME-XML)
- with this PR included, all generated OME-TIFF have `SamplesPerPixel` set to 3 as expected
- running Bio-Formats `showinf` utility on each OME-TIFF confirms the reader detects the data as RGB
- the converted APNG samples render fine but the other samples (multi RGB channels and multi-Z RGB) render incorrectly when compared to the ground truth
  - original
  <img width="1648" alt="Screenshot 2022-11-25 at 11 24 09" src="https://user-images.githubusercontent.com/1355463/203990212-a0651b2a-8ebf-4c53-8a88-37e9a2eda464.png">
  
  <img width="1504" alt="Screenshot 2022-11-25 at 11 23 46" src="https://user-images.githubusercontent.com/1355463/203990355-18a7c6c5-7893-44c3-9f2a-6d46d68978ed.png">

  - converted with this PR

   <img width="1648" alt="Screenshot 2022-11-25 at 11 22 35" src="https://user-images.githubusercontent.com/1355463/203990188-38fc3d9b-6090-48f3-9f08-a558c3b04a18.png">
  <img width="1504" alt="Screenshot 2022-11-25 at 11 23 01" src="https://user-images.githubusercontent.com/1355463/203990366-a471d4e8-2671-487f-98ad-fe5b1ab45e45.png">

melissalinkert commented:
I believe the display differences between the original and converted data are actually due to a bug in `MinMaxCalculator`, not a difference in the actual stored pixel values.

I could confirm that `showinf -autoscale -fast` on the original and converted data showed different images. The simple test code in https://gist.github.com/melissalinkert/5295a1237b93f52e88c324ddb89f3e5 shows that the pixel values between the two files are the same, as are the scaled pixel values when fixed min/max values are used for scaling. Inspecting the min/max values returned by `MinMaxCalculator` showed an unexpected difference though.

`showinf -autoscale -fast` with the changes in https://github.com/ome/bioformats/pull/3915 should show the same thing for both original and converted OME-TIFF files here.

sbesson commented:
Confirmed that the incorrect multi RGB channels rendering was fully related to the Bio-Formats utility issue. With the fix mentioned in https://github.com/ome/bioformats/pull/3915, the image now renders as the original.

For the multi-Z RGB image however, I think there is still a problem with the converted as the planes are written according in XYZCT order. The intermediate Zarr file displays correctly e.g. when opened in `napari`.

melissalinkert commented:
I think f532408 fixes the multi-Z + RGB case, so this is ready for review again.
